File: Binder/binder_server/src/main/java/com/hx/binder_server/MainActivity.java
package com.hx.binder_server;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public static void showlog(String info) {
        System.out.print("Watson "+info+"\n");
    }

}

File: Binder/messenger_server/src/main/java/com/gl/messenger_server/MainActivity.kt
package com.gl.messenger_server

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.util.Log
import androidx.appcompat.widget.AppCompatTextView
import com.gl.messenger_server.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }


    private val tv by lazy { AppCompatTextView(this) }

    private val handler = Handler(Looper.getMainLooper())

    private fun log(info: String) {
        Log.i("sample-allens", info)
        handler.post {
            tv.append(info)
            tv.append("\n")
        }
    }
}